python/pyarrow/parquet/core.py

```diff
@@ -1165,8 +1165,10 @@ def get_metadata(self):
         -------
         metadata : FileMetaData
         """
-        with self.open() as parquet:
-            return parquet.metadata
+        parquet = self.open()
+        meta = parquet.metadata
+        parquet.close(True)
```

Comment:
Hmm, isn't it a bug if a `with` statement fails to close the file it just opened?

Reply:
It's slightly awkward since ParquetFile distinguishes between opening and closing based on if it was the one who opened it. Indeed, `open()` opens the source for `ParquetFile` and thus should probably mark it as being responsible for closing. :+1: https://github.com/apache/arrow/pull/14729/commits/e845fc85c99ed3e484a320b48234c0dcf861a19e